www/apps/book/app/advanced-development/modules/db-operations/page.mdx

```diff
@@ -0,0 +1,482 @@
+    super(...arguments)
+    this.baseRepository_ = baseRepository
+  }
+}
+
+export default HelloModuleService
+```
+  </CodeTab>
+  <CodeTab label="Without Service Factory" value="no-service-factory">
+
+```ts highlights={[["10"]]}
+import { DAL } from "@medusajs/types"
+
+type InjectedDependencies = {
+  baseRepository: DAL.RepositoryService
+}
+
+class HelloModuleService {
+  protected baseRepository_: DAL.RepositoryService
+
+  constructor({ manager }: InjectedDependencies) {
+    this.baseRepository_ = baseRepository
+  }
+}
+
+export default HelloModuleService
+```
+
+  </CodeTab>
+</CodeTabs>
+
+Then, add the following method that uses it:
+
+export const repoHighlights = [
+  ["20", "transaction", "Wrap the function parameter in a transaction."]
+]
+
+```ts highlights={repoHighlights}
+// ...
+import { 
+  InjectManager,
+  InjectTransactionManager,
+  MedusaContext
+} from "@medusajs/utils"
+import { Context } from "@medusajs/types"
+import { EntityManager } from "@mikro-orm/knex"
+
+class HelloModuleService {
+  // ...
+  @InjectTransactionManager()
+  protected async update_(
+    input: {
+      id: string,
+      name: string
+    },
+    @MedusaContext() sharedContext?: Context<EntityManager>
+  ): Promise<any> {
+    return await this.baseRepository_.transaction(
+      async (transactionManager) => {
+        await transactionManager.nativeUpdate(
+          "my_custom",
+          {
+            id: input.id
+          },
+          {
+            name: input.name
+          }
+        )
+
+        // retrieve again
+        const updatedRecord = await transactionManager.execute(
+          `SELECT * FROM my_custom WHERE id = '${input.id}'`
+        )
+
+        return updatedRecord
+      }
+    )
+  }
+
+  @InjectManager()
+  async update(
+    input: {
+      id: string,
+      name: string
+    },
+    @MedusaContext() sharedContext?: Context<EntityManager>
+  ) {
+    return await this.update_(input, sharedContext)
+  }
+}
+```
+
+The `update_` method uses the `baseRepository_.transaction` method to wrap a function in a transaction.
+
+The function receives a transactional entity manager as a parameter. Use it to perform the database operations.
+
+<Note>
+
+Refer to [MikroORM's reference](https://mikro-orm.io/api/5.9/knex/class/EntityManager) for a full list of the entity manager's methods.
+
+</Note>
+
+### Configuration Parameter
+
+The `transaction` method of the `baseRepository` accepts an optional second parameter to configure the transaction.
+
+It's an object having the following properties:
+
+1. `transaction`: Set the transactional entity manager passed to the function. This is useful if your method must be a nested transaction:
+
+```ts highlights={[["16"]]}
+class HelloModuleService {
+  // ...
+  @InjectTransactionManager()
+  async update_(
+    input: {
+      id: string,
+      name: string
+    },
+    @MedusaContext() sharedContext?: Context<EntityManager>
```

Comment:
```suggestion
    @MedusaContext() sharedContext?: Context<SqlEntityManager> // from postgres
```

Reply:
Actually I checked and `SqlEntityManager` from `@mikro-orm/knex` is the same as importing `EntityManager` from `@mikro-orm/knex`, but maybe it's better to include the import in all code snippets to avoid confusion.